src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,178 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { SearchParams, getSearchWith } from '../utils/searchHelper';
+import { PersonSort } from '../types/PersonSort';
 
 export const PeoplePage = () => {
+  const [isLoader, setIsLoader] = useState(false);
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || '';
+  const century = searchParams.getAll('century') || [];
+  const query = searchParams.get('query') || '';
+  const sortFild = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  useEffect(() => {
+    setIsError(false);
+    setIsLoader(true);
+    getPeople()
+      .then(setPeople)
+      .catch((error) => {
+        setIsError(true);
+        throw error;
+      })
+      .finally(() => {
+        setIsLoader(false);
+      });
+  }, []);
+
+  const setSearchWith = (params: SearchParams) => {
+    const search = getSearchWith(searchParams, params);
+
+    setSearchParams(search);
+  };
+
+  const peopleFilter = useMemo(() => {
```

Comment:
let's use more self-descriptive name here
```suggestion
  const filteredPeople = useMemo(() => {
```